service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java

```diff
@@ -175,9 +178,26 @@ private void setSubject(Encounter encounter, POCDMT000002UK01ClinicalDocument1 c
             encounterParticipantComponents.add(dataEntererMapper
                 .mapDataEntererIntoParticipantComponent(clinicalDocument.getDataEnterer()));
         }
+
+        responsibleParty.ifPresent(it ->
+            encounterParticipantComponents.add(buildParticipantComponent(it, RESPONSIBLE_PARTY_PARTICIPANT_CODE,
+                RESPONSIBLE_PARTY_PARTICIPANT_DISPLAY))
+        );
+
         return encounterParticipantComponents;
     }
 
+    private EncounterParticipantComponent buildParticipantComponent(PractitionerRole it, String code, String display) {
```

Comment:
Maybe `PractitionerRole` argument could be named differently?

Reply:
Good point. Done.